update test-rsa-pss-sigs-on-certificate-verify.py for TLS1.3

### System Information

| Type                  | Version/Name |
| --------------------- | ------------ |
| Operating System Name | Ubuntu |
| Operating System Version | trusty 14.04.5  (Travis) |
| Python version        | Python 3.5.5  |
| tlslite-ng version    | 0.8.0-alpha3 |

### Problem description
`test-rsa-pss-sigs-on-certificate-verify.py` fails when run against tlslite-ng

but because in the Travis-CI only a subset of tests in `test-rsa-pss-sigs-on-certificate-verify.py` is run (through the `tlslite-ng-random-subset.json`), the issue is masked.

(Either the runner, or the `tlslite-ng-random-subset.json` script, should be updated so that this case is not skipped randomly)

### Describe how to reproduce the problem
```
python tests/scripts_retention.py tests/tlslite-ng.json `which tls.py`
```

### Include errors or backtraces

```
test-rsa-pss-sigs-on-certificate-verify.py:stdout:check CertificateRequest sigalgs ...
test-rsa-pss-sigs-on-certificate-verify.py:stdout:Error encountered while processing node <tlsfuzzer.expect.ExpectCertificateRequest object at 0x7f330e950668> (child: <tlsfuzzer.expect.ExpectServerHelloDone object at 0x7f330e9506a0>) with last message being: <tlslite.messages.Message object at 0x7f330e377240>
test-rsa-pss-sigs-on-certificate-verify.py:stdout:Error while processing
test-rsa-pss-sigs-on-certificate-verify.py:stdout:Traceback (most recent call last):
test-rsa-pss-sigs-on-certificate-verify.py:stdout:  File "scripts/test-rsa-pss-sigs-on-certificate-verify.py", line 486, in main
test-rsa-pss-sigs-on-certificate-verify.py:stdout:    runner.run()
test-rsa-pss-sigs-on-certificate-verify.py:stdout:  File "/home/travis/build/tomato42/tlsfuzzer/tlsfuzzer/runner.py", line 212, in run
test-rsa-pss-sigs-on-certificate-verify.py:stdout:    node.process(self.state, msg)
test-rsa-pss-sigs-on-certificate-verify.py:stdout:  File "/home/travis/build/tomato42/tlsfuzzer/tlsfuzzer/expect.py", line 697, in process
test-rsa-pss-sigs-on-certificate-verify.py:stdout:    self.sig_algs)
test-rsa-pss-sigs-on-certificate-verify.py:stdout:AssertionError: Unexpected sig algs. Got: [(8, 6), (8, 11), (8, 5), (8, 10), (8, 4), (8, 9), (6, 1), (5, 1), (4, 1), (3, 1), (2, 1)], expected: [(8, 6), (8, 5), (8, 4), (6, 1), (5, 1), (4, 1), (3, 1), (2, 1)]
```

--- a/scripts/test-rsa-pss-sigs-on-certificate-verify.py
+++ b/scripts/test-rsa-pss-sigs-on-certificate-verify.py
@@ -43,8 +43,9 @@ def help_msg():
     print("                names and not all of them, e.g \"sanity\"")
     print(" -e probe-name  exclude the probe from the list of the ones run")
     print("                may be specified multiple times")
-    print(" -n num         only run `num` random tests instead of a full set")
-    print("                (excluding \"sanity\" tests)")
+    print(" -n num         only run `num` long random fuzzing tests")
+    print("                instead of a full set")
+    print("                (excluding \"sanity\" and other short tests)")
     print(" -s sigalgs     signature algorithms expected in")
     print("                CertificateRequest. Format is either the new")
     print("                signature scheme (rsa_pss_sha256) or old one")
@@ -113,6 +114,7 @@ def main():
         raise ValueError("Specify certificate file using -c")
 
     conversations = {}
+    conversations_long = {}
 
     conversation = Connect(host, port)
     node = conversation
@@ -346,7 +348,7 @@ def main():
                                               AlertDescription.decrypt_error))
             node.next_sibling = ExpectClose()
             node = node.add_child(ExpectClose())
-            conversations["malformed rsa-pss in CertificateVerify - "
+            conversations_long["malformed rsa-pss in CertificateVerify - "
                           "xor {1} at {0}"
                           .format(pos, hex(xor))] = conversation
 
@@ -464,14 +466,15 @@ def main():
     bad = 0
     failed = []
     if not num_limit:
-        num_limit = len(conversations)
+        num_limit = len(conversations_long)
 
     # make sure that sanity test is run first and last
     # to verify that server was running and kept running throught
     sanity_test = ('sanity', conversations['sanity'])
     ordered_tests = chain([sanity_test],
-                          islice(filter(lambda x: x[0] != 'sanity',
-                                        conversations.items()), num_limit),
+                          filter(lambda x: x[0] != 'sanity',
+                                 conversations.items()),
+                          islice(conversations_long.items(), num_limit),
                           [sanity_test])
 
     for c_name, c_test in ordered_tests:

--- a/tests/tlslite-ng-random-subset.json
+++ b/tests/tlslite-ng-random-subset.json
@@ -204,10 +204,11 @@
                          "-c", "tests/clientX509Cert.pem"]
           },
          {"name" : "test-rsa-pss-sigs-on-certificate-verify.py",
-          "comment" : "test script has over 2k test cases, limit it",
+          "comment" : "test script has over 2k test cases, limit it; also, we expect specific sigalgs after RSA-PSS support was added to tlslite-ng",
           "arguments" : ["-n", "15",
                          "-k", "tests/clientX509Key.pem",
-                         "-c", "tests/clientX509Cert.pem"]
+                         "-c", "tests/clientX509Cert.pem",
+                         "-s", "8+6 8+11 8+5 8+10 8+4 8+9 6+1 5+1 4+1 3+1 2+1"]
           },
          {"name" : "test-extended-master-secret-extension-with-client-cert.py",
           "arguments" : ["-k", "tests/clientX509Key.pem",

--- a/tests/tlslite-ng.json
+++ b/tests/tlslite-ng.json
@@ -195,8 +195,10 @@
                          "-c", "tests/clientX509Cert.pem"]
           },
          {"name" : "test-rsa-pss-sigs-on-certificate-verify.py",
+          "comment" : "we expect specific sigalgs after RSA-PSS support was added to tlslite-ng",
           "arguments" : ["-k", "tests/clientX509Key.pem",
-                         "-c", "tests/clientX509Cert.pem"]
+                         "-c", "tests/clientX509Cert.pem",
+                         "-s", "8+6 8+11 8+5 8+10 8+4 8+9 6+1 5+1 4+1 3+1 2+1"]
           },
          {"name" : "test-extended-master-secret-extension-with-client-cert.py",
           "arguments" : ["-k", "tests/clientX509Key.pem",

